cmd/serve/main.go

```diff
@@ -1,22 +1,53 @@
 package main
 
 import (
+	"context"
 	"embed"
 	"fmt"
 	"io/fs"
 	"log"
 	"net/http"
+	"os"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+
+	create_user_v1 "github.com/sunb26/xat/handler/create_user"
 )
 
 //go:embed all:web all:web/_next
 var content embed.FS
 
+type middleware struct {
+	handler http.Handler
+	db      *sqlx.DB
+}
+
+func (i *middleware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := context.WithValue(r.Context(), "db", i.db)
+	i.handler.ServeHTTP(w, r.WithContext(ctx))
+}
+
+func newMiddleware(handler http.Handler, db *sqlx.DB) *middleware {
+	return &middleware{handler: handler, db: db}
+}
+
 func main() {
 	content, err := fs.Sub(content, "web")
 	if err != nil {
 		log.Fatal(err)
 	}
-	http.Handle("/", http.FileServerFS(content))
+
+	db, err := sqlx.Open("postgres", os.Getenv("DSN"))
+	if err != nil {
+		log.Fatalf("failed to open to database: %v", err)
+	}
+
+	mux := http.NewServeMux()
+
+	mux.Handle("/api/v1/user", newMiddleware(http.HandlerFunc(create_user_v1.CreateUser), db))
```

Comment:
still wrap the mux with the middleware, just register the static file server with another mux. then merge the muxes to be put in listen and serve

Reply:
it's going to get noisy real quickly `loggingMiddleware(middlewareX(middlewareY(...(http.HandlerFunc(...)...)))`, common practice to apply middleware to groups of endpoints